python/ml4ir/base/tests/test_monte_carlo_scorer.py

```diff
@@ -0,0 +1,49 @@
+import unittest
+from unittest.mock import MagicMock
+import tensorflow as tf
+import pathlib
+import yaml
+from ml4ir.base.model.scoring.monte_carlo_scorer import MonteCarloScorer
+
+
+class TestMonteCarloScorer(unittest.TestCase):
+
+    def setUp(self):
+        model_config_file = pathlib.Path(__file__).parent / "data" / "configs" / "model_config_monte_carlo_10.yaml"
+
+        with open(model_config_file.as_posix(), 'r') as f:
+            model_config = yaml.safe_load(f)
+
+        self.scorer = MonteCarloScorer(
+            model_config=model_config,
+            feature_config=MagicMock(),
+            interaction_model=MagicMock(),
+            loss=MagicMock(),
+            file_io=MagicMock()
+        )
+
+        self.scorer.architecture_op = MagicMock()
+
+    def test_10_calls(self):
+        inputs = {"f0": tf.constant([0.5], dtype=tf.float32)}
+
+        # Call the method
+        result = MonteCarloScorer.call(self.scorer, inputs, training=None)
+
+        # Check that the call method has looped the correct number of times
+        self.assertEqual(str(result["score"].name).count("iadd"), self.scorer.monte_carlo_inference_trials)
```

Comment:
@mohazahran Can you explain the logic behind what's happening here? I haven't seen this before myself.

Reply:
Why not just do something like this? This is more common unit testing practice, right?

```
import unittest
from unittest.mock import patch

class B:
    def super_method(self):
        pass

class A(B):
    def method(self):
        # Calls the super method
        self.super_method()

class TestA(unittest.TestCase):
    def test_super_method_call_count(self):
        with patch('path.to.module.B.super_method') as mock_super_method:
            a = A()
            a.method()
            # Call the method as many times as needed
            # ...

            # Assert the super method was called the expected number of times
            self.assertEqual(mock_super_method.call_count, 1)

if __name__ == '__main__':
    unittest.main()
```

Reply:
The unit test should verify number of times the `RelevanceScorer.call` was called IMO.

Reply:
This is basically checks the number of "addition" operation used in aggregating the MC trials which should be equals to number of MC trials.